jena-arq/src/main/java/org/apache/jena/sparql/expr/E_IRI.java

```diff
@@ -74,14 +74,15 @@ protected NodeValue evalSpecial(Binding binding, FunctionEnv env) {
 
     /*package*/ static NodeValue resolve(NodeValue relative, String baseIRI, FunctionEnv env) {
         if ( baseIRI == null ) {
-            // Legacy
             if ( env.getContext() != null ) {
                 Query query = (Query)env.getContext().get(ARQConstants.sysCurrentQuery);
                 if ( query != null )
                     baseIRI = query.getBaseURI();
+                // If still null, NodeFunctions.iri will use the system base.
+//                if ( baseIRI == null )
+//                    baseIRI = IRIs.getBaseStr();
```

Comment:
I think these two last lines can be removed? Looks like not needed anymore?

Reply:
It's not necessary, it does document an assumption about the behaviour of NodeFunctions.iri.